srai/loaders/osm_loaders/osm_tile_loader.py

```diff
@@ -0,0 +1,127 @@
+"""
+OSM tile loader.
+
+This module implements downloading tiles from given OSM tile server.
+"""
+from io import BytesIO
+from pathlib import Path
+from typing import Any, Callable, Optional, Union
+from urllib.parse import urljoin
+
+import pandas as pd
+import requests
+from PIL import Image
+
+from srai.regionizers.slippy_map_regionizer import SlippyMapRegionizer
+from srai.utils import geocode_to_region_gdf
+
+from .osm_tile_data_collector import DataCollector, InMemoryDataCollector, SavingDataCollector
+
+
+class TileLoader:
+    """
+    Tile Loader.
+
+    Downloads raster tiles from user specified tile server, like listed in [1]. Loader founds x, y
+    coordinates [2] for specified area and downloads tiles. Address is built with schema
+    {tile_server_url}/{zoom}/{x}/{y}.{resource_type}
+
+    References:
+        1. https://wiki.openstreetmap.org/wiki/Raster_tile_providers
+        2. https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames
+    """
+
+    def __init__(
+        self,
+        tile_server_url: str,
+        zoom: int,
```

Comment:
Maybe put in sensible defaults? It would make using this module easier and you wouldn't have to know or look for some weird URLs